rust-src/concordium_base/src/base.rs

```diff
@@ -0,0 +1,1239 @@
+            .map_err(serde::ser::Error::custom)?;
+        SerdeSerialize::serialize(&decimal, ser)
+    }
+}
+
+#[derive(Clone, PartialEq, Debug, Error)]
+/// An error that may be raised by converting from a
+/// [`Decimal`](rust_decimal::Decimal) to a [`PartsPerHundredThousands`].
+pub enum ConvertPartsPerHundredThousandsError {
+    #[error("Parts per thousand should not have more than 5 decimals.")]
+    TooManyDecimals,
+    #[error("Parts per thousand should not be negative.")]
+    Negative,
+    #[error("Parts per thousand out of bounds.")]
+    OutOfBounds,
+    #[error("Scale out of bounds.")]
+    ScaleError {
+        #[from]
+        inner: rust_decimal::Error,
+    },
+}
+
+impl TryFrom<rust_decimal::Decimal> for PartsPerHundredThousands {
+    type Error = ConvertPartsPerHundredThousandsError;
+
+    fn try_from(value: rust_decimal::Decimal) -> Result<Self, Self::Error> {
+        let mut f = value;
+        f.normalize_assign();
+        if f.scale() > 5 {
+            return Err(ConvertPartsPerHundredThousandsError::TooManyDecimals);
+        }
+        if !f.is_sign_positive() && !f.is_zero() {
+            return Err(ConvertPartsPerHundredThousandsError::Negative);
+        }
+        f.rescale(5);
+        if f.mantissa() > 100_000 {
+            return Err(ConvertPartsPerHundredThousandsError::OutOfBounds);
+        }
+        Ok(PartsPerHundredThousands {
+            parts: f.mantissa() as u32,
+        })
+    }
+}
+
+impl FromStr for PartsPerHundredThousands {
+    type Err = ConvertPartsPerHundredThousandsError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let decimal: rust_decimal::Decimal = s.parse()?;
+        Self::try_from(decimal)
+    }
+}
+
+impl<'de> SerdeDeserialize<'de> for PartsPerHundredThousands {
+    fn deserialize<D: serde::Deserializer<'de>>(des: D) -> Result<Self, D::Error> {
+        let f: rust_decimal::Decimal =
+            SerdeDeserialize::deserialize(des).map_err(serde::de::Error::custom)?;
+        let parts = PartsPerHundredThousands::try_from(f).map_err(serde::de::Error::custom)?;
+        Ok(parts)
+    }
+}
+
+impl SerdeSerialize for MintRate {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer, {
+        let x = rust_decimal::Decimal::try_new(self.mantissa.into(), self.exponent.into())
+            .map_err(serde::ser::Error::custom)?;
+        SerdeSerialize::serialize(&x, serializer)
+    }
+}
+
+impl FromStr for MintRate {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let f: rust_decimal::Decimal = s.parse()?;
+        f.try_into()
+    }
+}
+
+impl TryFrom<rust_decimal::Decimal> for MintRate {
+    type Error = anyhow::Error;
+
+    fn try_from(mut value: rust_decimal::Decimal) -> Result<Self, Self::Error> {
+        // FIXME: exponents will only be 28 at most for this type, so it is not entirely
```

Comment:
fixme

Reply:
I can't really fix this now,  and it is not really an issue in practice. So regrettably I will postpone this. This has been copied over from the rust-sdk, so it is not a new issue.